enclave-manager/server/server.go

```diff
@@ -158,29 +158,31 @@ func (c *WebServer) GetServiceLogs(
 	req *connect.Request[kurtosis_engine_rpc_api_bindings.GetServiceLogsArgs],
 	str *connect.ServerStream[kurtosis_engine_rpc_api_bindings.GetServiceLogsResponse],
 ) error {
-
-	result, err := (*c.engineServiceClient).GetServiceLogs(ctx, req)
+	engineClient, err := (*c.engineServiceClient).GetServiceLogs(ctx, req)
 	if err != nil {
 		return err
 	}
 
-	logs := getServiceLogsFromEngine(result)
-	for {
+	for engineClient.Receive() {
 		select {
 		case <-ctx.Done():
-			err := result.Close()
+			logrus.Debug("Closing the engine client after receiving abort signal from the client")
+			err := engineClient.Close()
 			if err != nil {
 				logrus.Errorf("Error ocurred: %+v", err)
+				return err
 			}
-			close(logs)
 			return nil
-		case resp := <-logs:
+		default:
+			resp := engineClient.Msg()
```

Comment:
Does this work when the stream is open but no new logs are available yet? @tedim52

Reply:
Yep I just tested it. Good callout too - I looked into how [`Receive()`](https://pkg.go.dev/connectrpc.com/connect#ServerStreamForClient.Receive) is working - it only evaluates to false if an error occurs or stream is closed. So as long as streams open, that for loop will keep going, but I added a check afterwards to handle the error via `engineClient.Error()`.